cpmonitor/management/commands/cleanupimages.py

```diff
@@ -0,0 +1,212 @@
+        field_values = []
+
+        for model in models:
+            field_value = self._extract_value(field, model)
+
+            if field_value:
+                field_values.append(field_value)
+
+        return field_values
+
+    def _extract_value(self, field, model) -> str:
+        field_value = field.value_from_object(model)
+
+        if self._is_path_object(field_value):
+            return field_value.path
+        else:
+            return field_value
+
+    def _is_path_object(self, field_value) -> bool:
+        return not isinstance(field_value, str)
+
+    def _contains_text_field_file_path(self, field, field_value) -> bool:
+        if not self._is_domain_text_field(field):
+            return False
+
+        return len(self._extract_text_field_file_paths(field_value)) > 0
+
+    def _extract_text_field_file_paths(self, field_value) -> list[str]:
+        real_file_paths = []
+        potential_file_paths = re.findall(
+            r"!\[[^\]]*\]\((?P<file_path>.*?)\s*\)", field_value
+        )
+
+        for file_path in potential_file_paths:
+            if not self._is_url(file_path):
+                real_file_paths.append(file_path)
+
+        return real_file_paths
+
+    def _is_url(self, value) -> bool:
+        validate_url = URLValidator()
+
+        try:
+            validate_url(value)
+        except ValidationError:
+            return False
+
+        return True
+
+    def _contains_file_field_file_path(self, field, field_value) -> bool:
+        if not self._is_domain_file_field(field):
+            return False
+
+        return len(self._extract_file_field_file_path(field_value)) > 0
+
+    def _is_domain_file_field(self, field) -> bool:
+        return self._is_domain_model(field) and isinstance(
+            field, DjangoModels.FileField
+        )
+
+    def _extract_file_field_file_path(self, field_value) -> list[str]:
+        return re.findall(r"".join(["^.*", IMAGES_PATH, ".*$"]), field_value)
+
+    def _resolve_image_paths_from_file_fields(self) -> list[str]:
+        file_paths = []
+
+        for model_class in apps.get_models():
+            models = model_class.objects.all()
+
+            for field in model_class._meta.fields:
+                if self._is_domain_file_field(field):
+                    file_paths += self._extract_file_paths_from_models(field, models)
+
+        return file_paths
+
+    def resolve_saved_images(self) -> list[str]:
+        saved_images = []
+
+        for sub_directory, _, images in os.walk(IMAGES_PATH):
+            for image in images:
+                if self._is_ignored(image, sub_directory):
+                    continue
+
+                relative_path = os.path.join(sub_directory, image)
+                saved_images.append(os.path.abspath(relative_path))
+
+        return saved_images
+
+    def _is_ignored(self, image, sub_directory) -> bool:
+        return ".gitkeep" in image or "test_database_uploads" in sub_directory
```

Comment:
I'm not happy with this solution but I didn't find any other. Maybe one of you have one?

Reply:
I think could could start in IMAGES_PATH/uploads then you do not need the .gitkeep check.
 "test_database_uploads"
1. This can be deleted on prod and testing as it is only used for local tests. (and even locally they shouldn't be deleted anyway if they are referenced in the database) 
2. There should be no "test_database_uploads" in the images folder. To use the test images you anyway would need to put them under uploads and can't distinguish them from the real ones.

Reply:
Good hint with the uploads directory, thanks. I've found a solution for the unit tests which is working with temporary directories, so the cpmonitor/images/uploads isn't used in the tests and the images in test_database_uploads will not be deleted anyway.

Reply:
I'm not sure I understood you correctly, but the command in your last comment is wrong. It creates this kind of structure:
![grafik](https://github.com/GermanZero-de/klimaschutzmonitor/assets/35843701/0fbf4629-70be-4ea6-8571-18bb6a188b2a)
But what @cafischer is talking about, and which shouldn't lead to the images being deleted (because the test data references them), is:
![grafik](https://github.com/GermanZero-de/klimaschutzmonitor/assets/35843701/14499d64-eada-4e34-b776-9dee95da3eda)
So the data from test_database_uploads has to go into the existing uploads directory, not next to it. Therefore the correct copy command would be this, with `/*` at the end of the source:
```bash
cp -r e2e_tests/database/test_database_uploads/* cpmonitor/images/uploads
```